src/common/runtime/server.ts

```diff
@@ -174,13 +163,28 @@ if (verbose) {
         return;
       }
 
+      const loadCasesPrefix = '/load?';
       const runPrefix = '/run?';
       const terminatePrefix = '/terminate';
 
-      if (request.url.startsWith(runPrefix)) {
+      if (request.url.startsWith(loadCasesPrefix)) {
+        const query = request.url.substr(loadCasesPrefix.length);
+        try {
+          const webgpuQuery = parseQuery(query);
+          const loader = new DefaultTestFileLoader();
+          for (const testcase of await loader.loadCases(webgpuQuery)) {
+            testcases.set(testcase.query.toString(), testcase);
+          }
+          response.statusCode = 200;
+          response.end();
+        } catch (err) {
+          response.statusCode = 500;
+          response.end(`load failed with error: ${err}\n${(err as Error).stack}`);
+        }
+      } else if (request.url.startsWith(runPrefix)) {
         const name = request.url.substr(runPrefix.length);
         try {
-          const testcase = (await testcases).get(name);
+          const testcase = testcases.get(name);
```

Comment:
Where does this get populated? Or can you only call `/run` after explicitly calling `/load`?

Reply:
> Or can you only call `/run` after explicitly calling `/load`?

Correct.

Reply:
Can we maybe add an error handler for that case? We check if the test case is found, but maybe also check if the `testcases` is empty and say forgot to call `load`, or something?

Reply:
I could, but I'm not sure it's worth it. This isn't a human API, and I'm not attempting to build a stable interface here. This is essentially an extension of the `dawn/node` runner, which just happens to live in the CTS code-base. I'd rather not go through the effort of pretending this is a stable API, when I fully intend to make breaking changes.

Reply:
Fair.